pkg/api/types.go

```diff
@@ -195,6 +195,119 @@ type VolumeSource struct {
 	NFS *NFSVolumeSource `json:"nfs"`
 }
 
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
```

Comment:
Some of the VolumeSource subtypes do not make sense as PersistentVolumeSources.  Maybe they will make sense eventually, but they don't now.  We had a decision to make.  We could either document this for each VolumeSource field and throw errors if used wrongly or we could split the concepts with a promise to merge them if it makes sense later.

Brian and I both came to the latter choice independently, but neither of us is doing the implementation.  It's entirely possible that we're both wrong (again, because neither of us is doing the implementation).  If you feel that way, I'm open to discussing it, but I really want to get something committed here - done is better than perfect.

We can always revisit this with very little consequence, I think (the API will be compatible).

So I'll put it back on you.  I am sympathetic to the argument that the duplication to support two structs and two sets of plugins is cumbersome.  Is it worth spending time on right now?  That's not a leading question - I am not the one tasked with getting this work committed :)